cirq/qis/states.py

```diff
@@ -321,6 +313,231 @@ def _infer_qid_shape_from_dimension(dim: int) -> Tuple[int, ...]:
     return (dim,)
 
 
+def infer_qid_shape(*states: 'cirq.QUANTUM_STATE_LIKE') -> Tuple[int, ...]:
+    """Infer the qid shape of a set of states.
+
+    This is a heuristic that is guaranteed to return a qid shape compatible
+    with all of the given states, but may fail (raising an error) even
+    if there is a unique compatible qid shape. If, after applying the heuristic,
+    the qid shape is ambiguous, then an error is raised unless the qid shape is
+    to be inferred from the shape of a state vector or density matrix. In that
+    case, if the dimension of the state space is a power of 2, then the state
+    space is assumed to be composed of qubits; otherwise, it is assumed to be
+    composed of a single qudit.
```

Comment:
the syntax of this docstring is hard to parse. It's written like .. backwards .. for computers instead of for humans.

"We use a heuristic to determine a qid shape compatible with all of the given shapes. The heuristic [insert brief description here]. If the dimension of state vector or density matrix (not tensors!) is a power of 2, then we assume qubits, otherwise it is assumed to be composed of a single qudit. If the qid shape is still ambiguous, an error is raised."

Reply:
I've rewritten it following your suggestion.